redfish-core/lib/assembly.hpp

```diff
@@ -0,0 +1,514 @@
+
+    // if there is assembly association, look
+    // for endpoints
+    crow::connections::systemBus->async_method_call(
+        [aResp,
+         chassisPath](const boost::system::error_code ec,
+                      const std::variant<std::vector<std::string>>& endpoints) {
+        if (ec)
+        {
+            BMCWEB_LOG_DEBUG("DBUS response error");
+            messages::internalError(aResp->res);
+            return;
+        }
+
+        const std::vector<std::string>* assemblyList =
+            std::get_if<std::vector<std::string>>(&(endpoints));
+
+        if (assemblyList == nullptr)
+        {
+            BMCWEB_LOG_DEBUG("No assembly found");
+            return;
+        }
+
+        std::vector<std::string> sortedAssemblyList = *assemblyList;
+        std::sort(sortedAssemblyList.begin(), sortedAssemblyList.end());
+
+        checkAssemblyInterface(aResp, chassisPath, sortedAssemblyList);
+        return;
+    },
+        "xyz.openbmc_project.ObjectMapper", assemblyPath,
+        "org.freedesktop.DBus.Properties", "Get",
+        "xyz.openbmc_project.Association", "endpoints");
+}
+
+/**
+ * @brief Api to check for assembly associations.
+ * @param[in] aResp - Shared pointer for asynchronous calls.
+ * @param[in] chassisPath - Chassis to which the assemblies are
+ * associated.
+ *
+ * @return None.
+ */
+inline void checkForAssemblyAssociations(
+    const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+    const std::string& chassisPath, const std::string& service)
+{
+    BMCWEB_LOG_DEBUG("Check for assembly association");
+
+    using associationList =
+        std::vector<std::tuple<std::string, std::string, std::string>>;
+
+    crow::connections::systemBus->async_method_call(
+        [aResp,
+         chassisPath](const boost::system::error_code ec,
+                      const std::variant<associationList>& associations) {
+        if (ec)
+        {
+            BMCWEB_LOG_DEBUG("DBUS response error");
+            messages::internalError(aResp->res);
+            return;
+        }
+
+        const associationList* value =
+            std::get_if<associationList>(&associations);
+        if (value == nullptr)
+        {
+            BMCWEB_LOG_DEBUG("DBUS response error");
+            messages::internalError(aResp->res);
+            return;
+        }
+
+        bool isAssmeblyAssociation = false;
+        for (const auto& listOfAssociations : *value)
+        {
+            if (std::get<0>(listOfAssociations) != "assembly")
+            {
+                // implies this is not an assembly
+                // association
+                continue;
+            }
+
+            isAssmeblyAssociation = true;
+            break;
+        }
+
+        if (isAssmeblyAssociation)
+        {
+            getAssemblyEndpoints(aResp, chassisPath);
+        }
+    },
+        service, chassisPath, "org.freedesktop.DBus.Properties", "Get",
+        "xyz.openbmc_project.Association.Definitions", "Associations");
```

Comment:
I think you mean getAssociationList(), right?

getAssociationsList() definition will get pulled from @deepakala-k 's changes [PR](https://github.com/ibm-openbmc/bmcweb/commit/92e72b3c6090a4305a335b4ed0257ee65e739c03) ,as part of which assembly.hpp will get updated.

Reply:
Sorry.  I meant - propose `getAssociationList()` to upstream,
or use `sdbusplus::asio::getProperty<AssociationList>(...){..}` instead of using `async_method_call()`